pycbc/events/significance.py

```diff
@@ -0,0 +1,253 @@
+                             "--far-calculation-method is 'trigger_fit'. "
+                             "Given as combination-values pairs, e.g. "
+                             "H1:0 L1:0 V1:-4, for all combinations which "
+                             "have --far-calculation-method "
+                             "of trigger_fit")
+    parser.add_argument("--fit-function", nargs='+', default=[],
+                        help="Functional form for the statistic slope fit if "
+                             "--far-calculation-method is 'trigger_fit'. "
+                             "Given as combination:function pairs, i.e. "
+                             "H1:exponential H1L1:n_louder H1L1V1:n_louder. "
+                             "Options: ["
+                             + ",".join(trstats.fitalpha_dict.keys()) + "]. "
+                             "Default = exponential for all")
+
+
+def check_significance_options(args, parser):
+    """
+    Check that the arguments given for the insert_significance_option_group
+    options make sense
+    """
+    # Check that the key:method/function/threshold are in the
+    # right format, and are in allowed combinations
+    lists_to_check = [(args.far_calculation_method, str,
+                       _significance_meth_dict.keys()),
+                      (args.fit_function, str,
+                       trstats.fitalpha_dict.keys()),
+                      (args.fit_threshold, float,
+                       range(-10000, 10000))]
+    # (10,000 is suitable for current use but can be changed if wanted)
+
+    for list_to_check, type_to_convert, allowed_values in lists_to_check:
+        key_list = []
+        for key_value in list_to_check:
+            try:
+                key, value = tuple(key_value.split(':'))
+            except ValueError:
+                parser.error("Need key:value format, got %s" % key_value)
+
+            if key in key_list:
+                parser.error("Duplicate key %s in a significance option" % key)
+            key_list.append(key)
+
+            try:
+                type_to_convert(value)
+            except ValueError:
+                err_fmat = "Value {} of key {} can't be converted"
+                parser.error(err_fmat.format(value, key))
+
+            if type_to_convert(value) not in allowed_values:
+                err_fmat = "Value {} of key {} is not in allowed values: {}"
+                parser.error(err_fmat.format(value, key, allowed_values))
+
+    # Are the functions/thresholds appropriate for the methods given?
+    methods = {}
+    # A method has been specified
+    for key_value in args.far_calculation_method:
+        key, value = tuple(key_value.split(':'))
+        methods[key] = value
+
+    # A function or threshold has been specified
+    function_or_thresh_given = []
+    for key_value in args.fit_function + args.fit_threshold:
+        key, _ = tuple(key_value.split(':'))
+        if key not in methods:
```

Comment:
This seems to assign a default value in what looks like a pretty opaque way.  If the function doesn't just check correctness but also assigns defaults, that should be noted somewhere IMO